corehq/apps/es/queries.py

```diff
@@ -0,0 +1,38 @@
+import re
+
+
+def match_all():
+    """
+    No-op query used because a default must be specified
+    """
+    return {"match_all": {}}
+
+
+def _smart_query_string(search_string):
+    special_chars = ['&&', '||', '!', '(', ')', '{', '}', '[', ']', '^', '"',
```

Comment:
Eh. that does look like a pretty nasty regex. Not sure this would be a readability win.

`re.compile('\&\&|\|\||\!|\(|\)|\{|\}|\[|\]|\^|\"|\~|\*|\?|\:|\\\\|\/')`

Could also do something like: 

```
a = re.compile('|'.join(map(lambda x: re.escape(x), special_chars)))
if a.search(search_string):
    return False, search_string
```